pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,404 @@ import (
-}
+	var tmpTestDir string
+	BeforeEach(func() {
+		tmpTestDir = testDir(os.TempDir())
+		if err != nil {
+			Fail(fmt.Sprintf("[BeforeEach] Failed created test dir: %v\n", err))
+		}
+		By(fmt.Sprintf("[BeforeEach] Creating temporary dir %s", tmpTestDir))
+		syscall.Umask(0000)
+		err = os.Mkdir(tmpTestDir, 0777)
+		if err != nil {
+			Fail(fmt.Sprintf("Could not create tmp file: %v\n ", err))
+		}
+	})
 
-func startHTTPServer(port int, dir string) (*http.Server, error) {
-	server := &http.Server{
-		Addr:    fmt.Sprintf(":%d", port),
-		Handler: http.FileServer(http.Dir(dir)),
-	}
+	AfterEach(func() {
+		By(fmt.Sprintf("[AfterEach] Cleaning up temporary dir %s", tmpTestDir))
+		os.RemoveAll(tmpTestDir)
+	})
 
-	go func() {
-		if err := server.ListenAndServe(); err != nil {
-			// cannot panic, because this probably is an intentional close
-			log.Printf("Httpserver: ListenAndServe() error: %s", err)
+	table.DescribeTable("when data is in a supported file format", func(originalFile string, useVirtSize bool, expectFormats []string) {
+		By(fmt.Sprintf("Getting size of source file %q", originalFile))
+		finfo, err := os.Stat(originalFile)
+		Expect(err).NotTo(HaveOccurred())
+		sourceSize := finfo.Size()
+		fmt.Fprintf(GinkgoWriter, "INFO: size = %d\n", sourceSize)
+
+		By(fmt.Sprintf("Converting source file to format: %s", expectFormats))
+		// Generate the expected data format from the random bytes
+		testSample, err := utils.FormatTestData(originalFile, tmpTestDir, expectFormats...)
+		Expect(err).NotTo(HaveOccurred(), "Error formatting test data.")
+		fmt.Fprintf(GinkgoWriter, "INFO: converted source file name is %q, in dir %q\n", testSample, tmpTestDir)
+
+		tempTestServer := createTestServer(filepath.Dir(testSample))
+		defer tempTestServer.Close()
+
+		testTarget := filepath.Join(tmpTestDir, common.ImporterWriteFile)
+		By(fmt.Sprintf("Importing %q to %q", tempTestServer.URL, testTarget))
+		err = CopyImage(testTarget, tempTestServer.URL+"/"+filepath.Base(testSample), "", "")
+		Expect(err).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("Checking size of the output file %q", testTarget))
+		var targetSize int64
+		if useVirtSize {
+			By("... using output image's virtual size")
+			targetSize = getImageVirtualSize(testTarget)
+			Expect(targetSize).To(Equal(sourceSize))
+		} else {
+			By("... using stat()")
+			finfo, err = os.Stat(testTarget)
+			Expect(err).NotTo(HaveOccurred())
+			targetSize = finfo.Size()
+			Expect(targetSize).To(Equal(sourceSize))
 		}
-	}()
-
-	started := false
-	for i := 0; i < 10; i++ {
-		resp, err := http.Get(fmt.Sprintf("http://localhost:%d/", port))
-		if err != nil {
-			time.Sleep(500 * time.Millisecond)
-			continue
+		fmt.Fprintf(GinkgoWriter, "INFO: byte size = %d\n", targetSize)
+
+		By(fmt.Sprintf("Calling `size.Size()` on same endpoint %q", tempTestServer.URL))
+		// extract the file extension(s) and check if file should be skipped
+		testBase := filepath.Base(testSample)
```

Comment:
Why not move this line up to 281.1 to save a call to Base()?

Reply:
Done